qwed_infra/guards/network_guard.py

```
@@ -89,38 +100,42 @@ def verify_reachability(self, resources: Dict[str, Any], source: str, destinatio
-            for sg_id in target_sgs:
-                rules = security_groups.get(sg_id, {}).get("ingress", [])
-                for rule in rules:
-                    rule_port = rule.get("port")
-                    port_match = (rule_port == port) or (rule_port == -1)
-                    if port_match:
-                        ingress_allowed = True
-                        break
-                if ingress_allowed:
+
+        # Determine the source IP for CIDR matching
+        source_ip = "0.0.0.0" if source == "internet" else source
```

Ruff (0.15.18)

[error] Possible binding to all interfaces

(S104)